src/Grid/grid.jl

```diff
@@ -665,27 +667,40 @@ end
 
 """
     getcoordinates!(x::Vector{Vec{dim,T}}, grid::AbstractGrid, cell::Int)
-Fills the vector `x` with the coordinates of a cell, defined by its cell id.
+    getcoordinates!(x::Vector{Vec{dim,T}}, grid::AbstractGrid, cell::AbstractCell)
+
+Fills the vector `x` with the coordinates of a cell defined by either its cellid or the cell object itself.
 """
-@inline function getcoordinates!(x::Vector{Vec{dim,T}}, grid::AbstractGrid, cell::Int) where {dim,T}
-    #@assert length(x) == N
+
+@inline function getcoordinates!(x::Vector{Vec{dim,T}}, grid::Ferrite.AbstractGrid, cellid::Int) where {dim,T} 
+    cell = getcells(grid, cellid)
+    getcoordinates!(x, grid, cell)
+end
+
+@inline function getcoordinates!(x::Vector{Vec{dim,T}}, grid::Ferrite.AbstractGrid, cell::Ferrite.AbstractCell) where {dim,T}
     @inbounds for i in 1:length(x)
-        x[i] = grid.nodes[grid.cells[cell].nodes[i]].x
+        x[i] = getcoordinates(getnodes(grid, cell.nodes[i]))
     end
+    return x
 end
+
 @inline getcoordinates!(x::Vector{Vec{dim,T}}, grid::AbstractGrid, cell::CellIndex) where {dim, T} = getcoordinates!(x, grid, cell.idx)
 @inline getcoordinates!(x::Vector{Vec{dim,T}}, grid::AbstractGrid, face::FaceIndex) where {dim, T} = getcoordinates!(x, grid, face.idx[1])
 
+# TODO: Deprecate one of `cellcoords!` and `getcoordinates!`, as they do the same thing
+cellcoords!(global_coords::Vector{Vec{dim,T}}, grid::AbstractGrid{dim}, i::Int) where {dim,T} = getcoordinates!(global_coords, grid, i) 
```

Comment:
This is only a rename, right? If so, then we can eliminate the todo by adding a deprecation binding to deprecations.jl and eliminate the todo. :)